strawberryfields/engine.py

```diff
@@ -653,26 +653,36 @@ def run(
             program, compile_options=compile_options, recompile=recompile, **kwargs
         )
         try:
-            while not job.finished:
-                if job.status == "failed":
-                    message = (
-                        f"The remote job {job.id} failed due to an internal "
-                        "server error. Please try again."
-                    )
-                    self.log.error(message)
-
-                    raise FailedJobError(message)
-
+            while True:
+                # TODO: needed to refresh connection; remove once xcc.Connection
+                # is able to refresh config info dynamically
+                job._connection = self.connection
+                job.clear()
+
+                if job.finished:
+                    break
                 time.sleep(self.POLLING_INTERVAL_SECONDS)
+
         except KeyboardInterrupt as e:
             xcc.Job(id_=job.id, connection=self.connection).cancel()
             raise KeyboardInterrupt("The job has been cancelled.") from e
 
-        if job.status == "complete":
+        if job.status == "failed":
+            message = (
+                f"The remote job {job.id} failed due to an internal "
+                "server error. Please try again."
+            )
```

Comment:
Don't forget to output the metadata here too!
```suggestion
            message = (
                f"The remote job {job.id} failed due to an internal "
                f"server error: {job.metadata}. Please try again."
            )
```